sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveScriptTransformationSuite.scala

```diff
@@ -36,6 +38,15 @@ class HiveScriptTransformationSuite extends BaseScriptTransformationSuite with T
 
   import ScriptTransformationIOSchema._
 
+  protected override def getTestResourcePath(fileName: String): String = {
+    val url = Thread.currentThread().getContextClassLoader.getResource(fileName)
+    // Copy to avoid URISyntaxException during accessing the resources in `sql/core`
+    val file = File.createTempFile("script-transformation-test", ".py")
+    file.deleteOnExit()
+    FileUtils.copyURLToFile(url, file)
+    file.getAbsolutePath
```

Comment:
Can we share the code? https://github.com/apache/spark/blob/a1e459ed9f6777fb8d5a2d09fda666402f9230b9/sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala#L60-L67

Reply:
> Can we share the code?
> 
> https://github.com/apache/spark/blob/a1e459ed9f6777fb8d5a2d09fda666402f9230b9/sql/hive-thriftserver/src/test/scala/org/apache/spark/sql/hive/thriftserver/ThriftServerQueryTestSuite.scala#L60-L67

Seems can, we this method fro SQLTestUtils, and BasicScripTransformationExecSuite extend SQLTestUtils

Reply:
So you mean we make it as a new interface?

Reply:
> Seems can, we this method fro SQLTestUtils, and 

How about adding a method for that in `SparkFunSuite`?
```
  // helper function
  protected final def getTestResourceFile(file: String): File = {
    new File(getClass.getClassLoader.getResource(file).getFile)
  }

  protected final def getTestResourcePath(file: String): String = {
    getTestResourceFile(file).getCanonicalPath
  }

  protected final def <<New Method> ...
```

Reply:
> How about adding a method for that in `SparkFunSuite`?

In this way, we need to change all `testFile`  method in `SQLQueryTestSuite`?

Reply:
I meant it like this;
```
// SparkFunSuite
  protected final def copyAndGetResourceFile(file: String): File = {...}

// ThriftServerQueryTestSuite
  override protected def testFile(fileName: String): String = {
    copyAndGetResourceFile(fileName).getAbsolutePath
  }

  // 
  test("SPARK-25990: TRANSFORM should handle schema less correctly (hive serde)") {
    assume(TestUtils.testCommandAvailable("python"))
    val scriptFilePath = copyAndGetResourceFile("test_script.py").getAbsolutePath
```

Reply:
Look perfetc, updated